lib/bridge/IrcBridge.js

```diff
@@ -102,6 +102,21 @@ function IrcBridge(config, registration) {
     );
 
     this._provisioner = null;
+
+    // Cache the mode of each channel, the visibility of each room and the
+    // known mappings between them. When any of these change, any inconsistencies
+    // should be resolved by keeping the matrix side as private as necessary
+    this._visibilityMap = {
```

Comment:
So I've gone through all the dependencies you're using in order to do this, and it's just:
 - A `DataStore`
 - A bot `MatrixClient`

Can we please yank all of this into its own file and dependency inject, so I don't have to constantly check to see if you're using other fun aspects of the `IrcBridge`? This would make it a lot clearer to follow, and avoids polluting `IrcBridge` more than it already is (poor thing).